src/components/ItemList.svelte

```diff
@@ -18,14 +20,22 @@
   export let totalItems;
   export let paginatedItems;
 
+  let nonExpiredItems = items.filter((item) => !isExpired(item.expires));
+  let shownItems = nonExpiredItems;
+
+  let showExpired = false;
+  let filterTerm = "";
+
   let currentPage = writable(1);
   setContext("currentPage", currentPage);
 
-  totalItems = items.length;
-  paginatedItems = chunk([...items], DEFAULT_ITEMS_PER_PAGE);
+  totalItems = shownItems.length;
+  paginatedItems = chunk([...shownItems], DEFAULT_ITEMS_PER_PAGE);
 
-  function filterItems(filterText) {
-    filteredItems = items.filter((item) => item.name.includes(filterText));
+  function filterItems(filterText, expiryCheck) {
+    shownItems = expiryCheck ? items : nonExpiredItems;
+    filterTerm = filterText;
+    filteredItems = shownItems.filter((item) => item.name.includes(filterText));
```

Comment:
I'm not sure about this function-- it's modifying global state, yet also takes parameters. I think in general its best if functions [don't use/modify global state](https://thevaluable.dev/global-variable-explained/) -- but I case see in this case that might be a bit complicated to refactor. If we're going to leave it as-is, I'd take out the function parameters, which are basically shadowing global state already.

```suggestion
  function filterItems() {
    shownItems = expiryCheck ? items : nonExpiredItems;
    filterTerm = filterText;
    filteredItems = shownItems.filter((item) => item.name.includes(filterText));
```